cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -306,6 +277,69 @@ Result<bool> IsSupportedParquetFile(const ParquetFileFormat& format,
 
 }  // namespace
 
+std::optional<compute::Expression> ParquetFileFragment::EvaluateStatisticsAsExpression(
+    std::shared_ptr<Field> field, std::shared_ptr<parquet::Statistics> statistics) {
```

Comment:
```suggestion
std::optional<compute::Expression> ParquetFileFragment::EvaluateStatisticsAsExpression(
    const Field& field, const parquet::Statistics& statistics) {
```

Minor nit: prefer accepting `const X&` over `std::shared_ptr<X>` or even `const std::shared_ptr<X>&` if you can.

Reply:
Made the change, thanks!